spec/services/claims/claim/remove_empty_mentor_training_hours_spec.rb

```diff
@@ -0,0 +1,33 @@
+require "rails_helper"
+
+describe Claims::Claim::RemoveEmptyMentorTrainingHours do
+  subject(:service) { described_class.call(claim:) }
+
+  let!(:claim) { create(:claim, reference: nil, status: :internal, school:) }
+  let(:school) { create(:claims_school, urn: "1234") }
+
+  it_behaves_like "a service object" do
+    let(:params) { { claim: } }
+  end
+
+  describe "#call" do
+    it "removes the mentor trainings without hours from a claim" do
+      create(:mentor_training, hours_completed: nil, claim:)
+      create(:mentor_training, hours_completed: nil, claim:)
+      training_with_hours = create(:mentor_training, hours_completed: 20, claim:)
+      expect { service }.to change(Claims::MentorTraining, :count).by(-2)
+
+      expect(claim.mentor_trainings).to eq([training_with_hours])
+    end
+
+    context "when mentor trainings have hours completed" do
+      it "does not remove mentor trainings with hours completed from a claim" do
+        training1 = create(:mentor_training, hours_completed: 10, claim:)
+        training2 = create(:mentor_training, hours_completed: 10, claim:)
+        expect { service }.to change(Claims::MentorTraining, :count).by(0)
+
+        expect(claim.mentor_trainings).to eq([training1, training2])
+      end
+    end
```

Comment:
I'm happy to drop this test. The case above shows:

1. It keeps mentor trainings _with_ hours completed.
2. It removes mentor trainings _without_ hours completed.
3. It handles the case where there are a mix of both complete and incomplete hours.

Not a blocker though.